addons/stock/models/stock_move.py

```diff
@@ -503,6 +503,9 @@ def _set_lot_ids(self):
                     move_line_vals['product_uom_id'] = move.product_id.uom_id.id
                     move_line_vals['qty_done'] = 1
                     move_lines_commands.append((0, 0, move_line_vals))
+                else:
+                    move_line = move.move_line_ids.filtered(lambda line: line.lot_id.id == lot.id)
+                    move_line['qty_done'] = 1
```

Comment:
```suggestion
                    move_line.qty_done = 1
```
the move line is already a record set